arangod/IResearch/IResearchLink.cpp

```diff
@@ -1530,33 +1592,32 @@ void IResearchLink::scheduleConsolidation(std::chrono::milliseconds delay) {
   task.async = _asyncFeature;
   task.id = id();
   task.state = _maintenanceState;
-  task.progress = [link = _asyncSelf.get()](){
+  task.progress = [link = _asyncSelf.get()]() {
     return !link->terminationRequested();
   };
 
-  ++_maintenanceState->pendingConsolidations;
+  _maintenanceState->pendingConsolidations.fetch_add(1, std::memory_order_release);
   task.schedule(delay);
 }
 
-Result IResearchLink::insert(
-    transaction::Methods& trx,
-    LocalDocumentId const& documentId,
-    velocypack::Slice const doc) {
+Result IResearchLink::insert(transaction::Methods& trx,
+                             LocalDocumentId documentId, velocypack::Slice doc) {
   TRI_ASSERT(_engine);
   TRI_ASSERT(trx.state());
 
   auto& state = *(trx.state());
 
-  if (_dataStore._inRecovery && _engine->recoveryTick() <= _dataStore._recoveryTick) {
+  if (_dataStore._inRecovery.load(std::memory_order_acquire) &&
+      _engine->recoveryTick() <= _dataStore._recoveryTick) {
     LOG_TOPIC("7c228", TRACE, iresearch::TOPIC)
-      << "skipping 'insert', operation tick '" << _engine->recoveryTick()
-      << "', recovery tick '" << _dataStore._recoveryTick << "'";
+        << "skipping 'insert', operation tick '" << _engine->recoveryTick()
+        << "', recovery tick '" << _dataStore._recoveryTick << "'";
 
     return {};
   }
 
-  auto insertImpl = [this, &trx, &doc, &documentId](
-                        irs::index_writer::documents_context& ctx) -> Result {
+  auto insertImpl = [this, doc, documentId,
+                     &trx](irs::index_writer::documents_context& ctx) -> Result {
```

Comment:
sometimes auto-formatting sucks :-)

Reply:
Fixed so it works for automatic formatting too